Allow BattleInfo to slightly move when Pokemon is targeted in double battles

francktrouillez opened the pull request:
## What are the changes?

This PR aims at allowing the battle info of the targeted enemy Pokemon to move during double battles. Currently, only the Pokemon itself allows to know the targeted enemy Pokemon, by continuously modifying the opacity. This PR aims at allowing the battle info to move as well.

For the context, I'm referencing this as a `BattleInfo`:
![image](https://github.com/pagefaultgames/pokerogue/assets/57403591/076bc03d-70e2-4e84-83b9-7f998cf95d60)


I was inspired by _Pokemon Emerald_ UI on this, where the battle info of the targeted enemy Pokemon moves as well (I actually took inspiration from this video on [Youtube](https://www.youtube.com/watch?v=DIu0hFzxzxQ)).

https://github.com/pagefaultgames/pokerogue/assets/57403591/c8cbb868-026a-41cc-abc2-426c6ef60d46

## Why am I doing these changes?
I'm a big fan of the Pokemon license, but I need to say I wasn't following all the new generations (I probably kinda stopped around gen 5 or 6). Because of that, during double battles, I was struggling a bit to understand which "battle info" I was targeting, and it happened more than once that I actually chose the wrong Pokemon because of my lack of Pokemon knowledge.

I believe that this change will help players like me to better understand which Pokemon they are targeting during double battles.

## What did change?

Before, the battle info of the targeted enemy Pokemon was never moving. The only way to know which Pokemon was targeted was by looking at the Pokemon itself, which was changing its opacity.

Now, the battle info moves as well when the enemy Pokemon is targeted. It will move up and down by a bit in a linear way to make it clear that the player is targeting this Pokemon (and the related battle info).

### Videos

Before:

https://github.com/pagefaultgames/pokerogue/assets/57403591/b5f2d885-abfd-4620-b337-2eefbabda5e7

After:

https://github.com/pagefaultgames/pokerogue/assets/57403591/9395cd52-2f85-4919-8d22-143a8cfe11e2

## How to test the changes?
I currently tested the changes manually, as I currently don't know how easy it would be to write actual tests for this. I'm open to any feedback on how to improve the testing process.

I've been using the [`src/overrides.ts`](https://github.com/pagefaultgames/pokerogue/blob/6952fe065b44c0c6371f0a575b8cf5fb8e97dbe3/src/overrides.ts) file to test the changes.

To do so, I just enabled double battles, and ensured to have an attacking move, an all-enemy focuses move, and a move affecting every other Pokemon on the field. I've been using `TACKLE`, `RAZOR_LEAF` and `SURF` for this purpose.
was changed to
```typescript
// ...
export const DOUBLE_BATTLE_OVERRIDE: boolean = true;
// ...
export const MOVESET_OVERRIDE: Array<Moves> = [Moves.TACKLE, Moves.RAZOR_LEAF, Moves.SURF];
// ...
```

Since this is just UI changes, I believe I'm not introducing any new bugs.

## Alternatives
<details>
  <summary>Before edit</summary>
  Currently, I'm making the box move in a fluent way. We can make it move in a more "jumpy" way, like in _Pokemon Emerald_, but I'm not sure it would fit the current UX/UI of the game, as the Pokemon itself is changing its opacity in a fluent way as well (as opposed to the "jumpy" way in Pokemon Emerald). I'm open to any feedback on this.
</details>

I currently make the box move up and down in a linear way. Does that look good from a UX perspective?

Of course, I'm open to any suggestion regarding this change, or about a way to make the double battles more user-friendly.

TempsRay commented:
Can you also show some other test cases like targeting your ally or using a move that hits multiple Pokemon? 


Code looks okay but personally I'm not a fan of the speed and way the battle info box bounces right now. Maybe someone else can offer their aesthetic input.

CodeTappert commented:
Aeastheticly it looks not good. Very much wobbly. Could you instead just make it move up/down?

francktrouillez commented:
@TempsRay 
> Can you also show some other test cases like targeting your ally or using a move that hits multiple Pokemon?

Sure, I updated the videos in the PR body to include the case where I target my ally. For moves that target multiple Pokemons, I'm not able to see the Pokemon focused beforehand even before. **I'm not aiming at adding this in this PR.**

Did I miss something? Is there any move where you can explicitly select multiple targets? I'd be glad to test them!

@TempsRay 
> Code looks okay but personally I'm not a fan of the speed and way the battle info box bounces right now. Maybe someone else can offer their aesthetic input.

@CodeTappert 
> Aeastheticly it looks not good. Very much wobbly. Could you instead just make it move up/down?

I made a change to only move up and down in a linear way. I updated the videos in the PR body to reflect that change. We can of course tweak the values such as the duration or the displacement, or even the ease function we use.

I'm open to suggestions. What do you think?

CodeTappert commented:
Looks way better!

Greenlamp2 reviewed (approved):
lgtm, nice QOL to help new players

bennybroseph commented:
@francktrouillez Seems like this issue is being reported related to this change [Discord Link](https://discord.com/channels/1125469663833370665/1243457704861896786/1243457704861896786)

francktrouillez commented:
@bennybroseph 
> @francktrouillez Seems like this issue is being reported related to this change [Discord Link](https://discord.com/channels/1125469663833370665/1243457704861896786/1243457704861896786)

Thanks for reporting the issue! I've proposed a fix in https://github.com/pagefaultgames/pokerogue/pull/1357.
Sorry for the inconvenience, I'll be more careful next time by testing more cases!
